### ISSUE TITLE
Add security operation ignores portfolio's base currency, hardcodes EUR by default

### ISSUE THREAD

**Issue by pfalcon** [buchen/portfolio#3444]
**To Reproduce**
1. Use File -> New to create a new portfolio file
2. Choose USD as the base currency
3. Use "New instrument..." dialog to add new security
4. Search for "AMZN"
5. Choose Yahoo Finance as the source feed.
6. After clicking "Apply" in search result list, security property dialog open and the currency set to EUR.

**Expected behavior**

As the portfolio base currency is set to USD, that what should be used by default for new securities (unless there're strong reasons to do otherwise, e.g. is source indicates that security's currency is different one. In this case we know that AMZN is quoted in USD.)

Version 0.64.4 (git master)

**Comment by pfalcon** [comment]
@buchen , I would suggest it's pretty important to fix this issue, because I may imagine it contributes to a perception that PortfolioPerformance is "intended for European investors" and "less suitable for US investors". (I read such characterizations more than once "on the internets".  Which is of course not true, but rather a subjective perception grounded in cases like above.)

The causes of this issue are apparently:
1. This line: https://github.com/buchen/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/model/Security.java#L50, i.e. hardcoding EUR by default in a Security.
2. And then, creating Security without explicitly setting currency in many places in the codebase.

@buchen, only you, as the author, would know the right way to fix this. E.g., the "ultimate" way would be to remove the default assignment above, disable default (empty) constructor, and make sure that any remaining constructor accepts currency explicitly. Then, fix the codebase to adhere. Yes, that would be a bit "revolutionary" (a lot of changes) way, so again, only you would know if it's worth it.

**Comment by pfalcon** [comment]
For a user who look to just work around the issue on their side, a low-effort way would be to change currency in Security.java. But they would be greeted by failing tests, because again, assumption of EUR is implicitly hardcoded in many places. I would like to contribute at least to resolving such cases: https://github.com/buchen/portfolio/pull/3445 . (But without resolving the root cause, such cases would reappear).

**Comment by buchen** [comment]
@pfalcon Agree. If the currency is not known otherwise (because it is imported via PDF or because it is know from the market), then it should be set to the base currency of the file.

The hard-coded "EUR" is a path dependency. The initial version of PP did not know anything about currencies. It just assumed all instruments and accounts use the same currency (EUR or CHF or USD or whatever). Then I introduced currencies, let the user choose during migration, but defaulted securities to EUR to just have a currency readily available. That reasoning was not sound.

I appreciate your proposal for a pull request. I understand the reasoning. If we remove the implicit constructor, then we have to choose the currency when creating a security - and make sure to use one from the client. But changing hundreds of tests also is a lot of changes.

I wonder if we annotate the default constructor with Guava's ```@VisibleForTesting```. Then at least SonarLint generates a warning in the IDE (see below). Typically, I do not fix all SonarLint warnings (too much for existing code), but whenever I am changing code, I look at the SonarLint warnings.

<img width="1154" alt="Bildschirmfoto 2023-07-15 um 22 37 26" src="https://github.com/buchen/portfolio/assets/587976/a30ac3d1-09f5-4479-aa9f-d3f32431734d">

Let me change the main places where new securities are created.
